cmd/api/src/api/v2/apiclient/ingest.go

```diff
@@ -71,6 +75,39 @@ func (s Client) SendFileUploadData(payload map[string]any, id int64) error {
 	}
 }
 
+func (s Client) SendCompressedFileUploadData(jsonFile io.Reader, id int64) error {
+	var (
+		body bytes.Buffer
+		gw   = gzip.NewWriter(&body)
+	)
+
+	if n, err := io.Copy(gw, jsonFile); err != nil {
```

Comment:
The json file is an embedded file living in `embed.FS` so I want to say the `io.Reader` in this case is just a `[]byte`. That is to say all the data is already living in memory. Because of that, I think @computator's proposed optimization would not have that huge of a performance impact.

That said, if we wanted to load data directly from disk in addition to what's in `embed.FS` then including the change would definitely grant us that flexibility.

That said, it's a small change but I don't think it's necessary if you'd rather not include it in this changeset.

Reply:
Using a pipe here could introduce a bit of complexity here to solve a problem that doesn't seem to exist currently. I think I will skip this change until we have a need for it.

Reply:
@computator FWIW AzureHound does not use this.